File: 0041-first-missing-positive/0041-first-missing-positive.py
from collections import Counter
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        mp = Counter(nums)

        for i in range(1 , max(nums)+1):
            if i not in mp:
                return i
        
        x = max(nums)
        
        if x < 0: return 1
        else : return x+1
        

        

File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        unordered_set<int> s;
        int x = 0;
        for (int i : nums){
            s.insert(i);
            x = max(x , i);
        }

        if (x < 0) return 1;

        for (int i = 1 ; i <= x ; i++){
            if (s.find(i)==s.end()) return i;
        }

        return x+1;





    }
};